torchmetrics/wrappers/bootstrapping.py

```diff
@@ -0,0 +1,174 @@
+            >>> output = bootstrap.compute()
+            >>> mean, std = output
+            >>> print(mean, std)
+            tensor(0.2205) tensor(0.0859)
+
+        """
+        super().__init__(compute_on_step, dist_sync_on_step, process_group, dist_sync_fn)
+        if not isinstance(base_metric, Metric):
+            raise ValueError(
+                "Expected base metric to be an instance of torchmetrics.Metric"
+                f" but received {base_metric}"
+            )
+
+        self.metrics = nn.ModuleList([deepcopy(base_metric) for _ in range(num_bootstraps)])
+        self.num_bootstraps = num_bootstraps
+
+        self.mean = mean
+        self.std = std
+        if quantile is not None and not _TORCH_GREATER_EQUAL_1_7:
+            raise ValueError('quantile argument can only be used with pytorch v1.7 or higher')
+        self.quantile = quantile
+        self.raw = raw
+
+        allowed_sampling = ('poisson', 'multinomial')
+        if sampling_strategy not in allowed_sampling:
+            raise ValueError(
+                f"Expected argument ``sampling_strategy`` to be one of {allowed_sampling}"
+                f" but recieved {sampling_strategy}"
+            )
+        self.sampling_strategy = sampling_strategy
+
+    def update(self, *args: Any, **kwargs: Any) -> None:
+        """ Updates the state of the base metric. Any tensor passed in will be bootstrapped along dimension 0 """
+        for idx in range(self.num_bootstraps):
+            args_sizes = apply_to_collection(args, Tensor, len)
+            kwargs_sizes = list(apply_to_collection(kwargs, Tensor, len))
+            if len(args_sizes) > 0:
+                size = args_sizes[0]
+            elif len(kwargs_sizes) > 0:
+                size = kwargs_sizes[0]
+            else:
+                raise ValueError('None of the input contained tensors, so could not determine the sampling size')
+            sample_idx = _bootstrap_sampler(size, sampling_strategy=self.sampling_strategy)
+            new_args = apply_to_collection(args, Tensor, torch.index_select, dim=0, index=sample_idx)
+            new_kwargs = apply_to_collection(kwargs, Tensor, torch.index_select, dim=0, index=sample_idx)
+            self.metrics[idx].update(*new_args, **new_kwargs)
+
+    def compute(self) -> List[Tensor]:
+        """ Computes the bootstrapped metric values. Allways returns a list of tensors, but the content of
+        the list depends on how the class was initialized
+        """
+        computed_vals = torch.stack([m.compute() for m in self.metrics], dim=0)
+        output = []
+        if self.mean:
+            output.append(computed_vals.mean(dim=0))
+        if self.std:
+            output.append(computed_vals.std(dim=0))
+        if self.quantile is not None:
+            output.append(torch.quantile(computed_vals, self.quantile))
+        if self.raw:
+            output.append(computed_vals)
+        return output
```

Comment:
this variant list can be tricky for interpretation, I would consider fix length and if some are not computed, leave it as None or NaN

Reply:
or return as a dictionary so the meaning also clear.. or is there a reason for this floating length?